cezo_fl/util/model_helpers.py

```diff
@@ -73,3 +80,30 @@ def get_trainable_model_parameters(
     for param in model.parameters():
         if param.requires_grad:
             yield param
+
+
+def model_forward(model: AllModel, batch_inputs: torch.Tensor | LLMBatchInput):
+    if isinstance(model, (OPTForCausalLM, PeftModel)):
+        assert isinstance(batch_inputs, LLMBatchInput)
+        return model(input_ids=batch_inputs.input_ids, attention_mask=batch_inputs.attention_mask)
+    elif isinstance(model, torch.nn.Module):
+        assert isinstance(batch_inputs, torch.Tensor)
+        return model(batch_inputs)
+    else:
+        raise Exception("This model type is not supported")
+
+
+def model_generate(model: LanguageModel, batch_inputs: LLMBatchInput, generation_kwargs: dict = {}):
```

Comment:
you should not use `generation_kwargs: dict = {}`, which is a common mistake in python. Even though I think the code here behaves correct, it is still dangerous.

it should be `generation_kwargs: dict | None = None` and inside the function we have `if generation_kwargs is None: generation_kwargs={}`. See the reason here: https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

Reply:
Thanks for pointing this out, resolved this issue by removing the default.
In our project so far, this is an required field for all cases.